hr_public_holidays/hr_public_holidays.py

```diff
@@ -87,37 +85,33 @@ def is_public_holiday(self, cr, uid, dt, employee_id=None, context=None):
                                employee.address_id.state_id.id)
                               ]
 
-        hr_holiday_public_line_obj = self.pool['hr.holidays.public.line']
-        holidays_line_ids = \
-            hr_holiday_public_line_obj.search(cr, uid, states_filter)
-        lines_obj = \
-            hr_holiday_public_line_obj.browse(cr, uid, holidays_line_ids)
-        for line in lines_obj:
+        hr_holiday_public_line_obj = self.env['hr.holidays.public.line']
+        holidays_line_ids = hr_holiday_public_line_obj.search(states_filter)
+        for line in holidays_line_ids:
             if date.strftime(dt, "%Y-%m-%d") == line.date:
                 return True
-
         return False
 
-    def get_holidays_list(self, cr, uid, year, employee_id=None,
-                          context=None):
+    def get_holidays_list(self, year, employee_id=None):
 
         res = []
-        employee = self.pool['hr.employee'].browse(cr, uid, employee_id)
+        employee = self.env['hr.employee'].browse(employee_id)
 
         holidays_filter = [('year', '=', year)]
         if not employee or not employee.address_id.country_id:
             holidays_filter.append(('country_id', '=', False))
         else:
-            holidays_filter += ['|', ('country_id', '=',
-                                      employee.address_id.country_id.id),
-                                ('country_id', '=', False)]
```

Comment:
The whole if block can be replaced with one line
holidays_filter.append(('country_id', 'in', list({False, employee.address_id.country_id.id}))